pkg/clients/gateway_client.go

```diff
@@ -121,18 +110,33 @@ func (gc *gatewayClient) RegisterSplicedStream(cc grpc.ClientConnInterface, name
 	})
 }
 
-func dial(ctx context.Context, address, id string, sharedKeys *keyring.SharedKeys, tlsConfig *tls.Config) (*grpc.ClientConn, error) {
+func dial(ctx context.Context, address, id string, kr keyring.Keyring, tlsConfig *tls.Config, lg *zap.SugaredLogger) (*grpc.ClientConn, error) {
+	authChallenge, err := authv2.NewClientChallenge(kr, id, lg)
+	if err != nil {
+		return nil, err
+	}
+
+	sessionAttrChallenge, err := session.NewClientChallenge(kr)
+	if err != nil {
+		return nil, err
+	}
+
+	challengeHandler := challenges.Chained(
+		authChallenge,
+		challenges.If(sessionAttrChallenge.HasAttributes).Then(sessionAttrChallenge),
+	)
 	return grpc.DialContext(ctx, address,
 		grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)),
-		grpc.WithChainStreamInterceptor(otelgrpc.StreamClientInterceptor(), cluster.NewClientStreamInterceptor(id, sharedKeys)),
+		grpc.WithChainStreamInterceptor(
```

Comment:
Order has changed between auth-interceptor implementations. Intended?

Reply:
After reviewing this again I think the previous order is actually better for tracing accuracy. :+1: